src/main.rs

```diff
@@ -48,8 +49,17 @@ fn virtualization_container() -> Result<bool, Error> {
 }
 
 fn main() {
+    let root: Cow<'static, str> =
+        env::var("ZRAM_GENERATOR_ROOT").map(|mut root| {
+            if !root.ends_with(path::is_separator) {
+                root.push('/');
+            }
+            println!("Using {:?} as root directory", root);
+            root.into()
```

Comment:
Hmm, https://doc.rust-lang.org/std/path/struct.Path.html#method.join seems to imply that `join` deals with trailing slashes correctly. So maybe this chunk will not be necessary if the number of places below where `root` is used is reduced?

Reply:
I think this was used only because of `{root}dev/something`, so this definitely can go now.